core/src/main/java/uk/gov/gchq/koryphe/impl/function/StringReplace.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.impl.function;
+
+import org.apache.commons.lang3.StringUtils;
+
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+/**
+ * A {@code StringReplace} is a {@link java.util.function.Function} which searches an input {@link String} for all
+ * occurrences of a search string, and replaces those matches with an alternative string.
+ */
+@Since("1.8.3")
+@Summary("Replace all portions of a string which match a regular expression.")
```

Comment:
This seems very similar to the StringRegexReplace Function. Do we need both?

Reply:
It is just slightly easier to use for noice users.